xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -117,6 +117,7 @@ SelectionStream& CSelectionStreams::Get(StreamType type, int index)
 
 std::vector<SelectionStream> CSelectionStreams::Get(StreamType type)
 {
+  CSingleLock lock(m_section);
```

Comment:
On 2016/03/25 08:26, Rainer Hochecker <notifications@github.com> wrote:
> VideoPlayer has its own thread and only this thread is supposed to access this method. It was designed only for use in OpenDefaultStreams: https://github.com/xbmc/xbmc/pull/700/commits/240692b2a3a4b9675bb4c972e745d7a1e8e1c3de

I see.  So that's the current state, which we must deal with.  Which
means my commit is necessary.

If you don't want my commit, plan B is to remove all uses of
CSelectionStreams from other parts of Kodi, then document
CSelectionStreams to be **not** thread-safe, and remove locking
overhead.

What do you want me to do?

> Unfortunately folks started to use it for different purpose like exposing information to the GUI. When calling this method the vector is copied twice which is a performance issue.

Twice?  It is copied once (i.e. data from "m_Streams" is copied to
local variable "streams").  The "return streams" is not a copy,
it's "moved" to the caller.  That is just a pointer exchange.  Some
C++ compilers might even be able to eliminate this move (maybe
gcc+clang do this already, I'm not sure).